contracts/cw4-voting/src/contract.rs

```diff
@@ -111,12 +113,16 @@ pub fn execute_member_changed_hook(
         } else {
             negative_difference += Uint128::from(old - weight);
         }
-        USER_WEIGHTS.save(
-            deps.storage,
-            &user_address,
-            &Uint128::from(weight),
-            env.block.height,
-        )?;
+        if weight == 0 {
```

Comment:
Could we add an explanatory comment here about why this works? Might be nice to add a similar one to `query_voting_power_at_height` just to make sure someone doesn't unknowingly remove the `unwrap_or_default` there.